### PR TITLE
[stable9] getShareByToken() should also consider remote shares

### PR DESCRIPTION
**Pull request by schiessle**
selective backport from master to fix https://github.com/nextcloud/server/issues/262

cc @LukasReschke

### CODE REVIEW

File: lib/private/share20/manager.php [hunk 1]
```diff
@@ -988,6 +988,18 @@ public function getShareByToken($token) {
 
 		$share = $provider->getShareByToken($token);
```
Comment (LukasReschke) [inline lib/private/share20/manager.php]: Doesn't this have to go? :)

Comment (schiessle) [inline lib/private/share20/manager.php]: yes :smile:

**Comment by LukasReschke** [comment]
Works. Awesome. Thanks a lot! 🚀

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Comment by schiessle** [comment]
seems like the lgtm bot doesn't like me :disappointed:

**Pull request merged**